packages/enzyme-matchers/src/utils/html.js

```diff
@@ -20,9 +20,13 @@ function mapWrappersHTML(wrapper: EnzymeObject): Array<string> {
       : inst._currentElement.props;
     consoleObject.error = error;
 
-    const transformedProps = Object.keys(props).map(
-      key => `${key}="${props[key]}"`,
-    );
+    const transformedProps = Object.keys(props).map(key => {
+      try {
+        return `${key}="${props[key].toString()}"`;
+      } catch (e) {
+        return `${key}="[object Object]"`;
+      }
```

Comment:
Interesting. I'm trying to think if this could breakdown in other situations..

Deos it not work to just do something like `typeof props === 'proxy'` ? (i assume that's not actually valid, but along those lines)